db/tx.go

```diff
@@ -149,6 +188,46 @@ func getWitnessTranfer(bz bazooka.Bazooka, DBI DB, tx core.Tx) (fromMerkleProof,
 	return fromMerkleProof, toMerkleProof, dbCopy, nil
 }
 
+// getWitnessMassMigration fetches the witness for transfer mass migrations
+// amongst other thins it also returns a DB transaction handler that can be used to rollback changes made to state tree while creating witness
+func getWitnessMassMigration(bz bazooka.Bazooka, DBI DB, tx core.Tx) (fromMerkleProof, toMerkleProof bazooka.StateMerkleProof, txDBConn DB, err error) {
+	dbCopy, _ := NewDB(bz.Cfg)
+
+	from, _, err := bz.FetchFromAndToStateIDs(tx)
+	if err != nil {
+		return
+	}
+
+	// fetch from state MP
+	err = DBI.fetchMPWithID(from, &fromMerkleProof)
+	if err != nil {
+		return
+	}
+
+	newFrom, _, err := bazooka.ApplyTx(bz, fromMerkleProof.State.Data, []byte(""), tx)
+	if err != nil {
+		return
+	}
+
+	mysqlTx := dbCopy.Instance.Begin()
+	defer func() {
+		if r := recover(); r != nil {
+			mysqlTx.Rollback()
+		}
+	}()
+	dbCopy.Instance = mysqlTx
+
+	// apply the new from leaf
+	currentFromStateCopy := fromMerkleProof.State
+	currentFromStateCopy.Data = newFrom
+	err = dbCopy.UpdateState(currentFromStateCopy)
+	if err != nil {
+		return
+	}
+
+	return fromMerkleProof, toMerkleProof, dbCopy, nil
```

Comment:
Where is this toMerkleProof come from? Or will it be just an empty data structure?

Reply:
It will be empty, just added so that it satisfies the interface, it isn't accessed anytime down the stack so should be okay